File: springboot3-snapshot-test/src/main/java/com/example/springbootsnapshottest/controller/HelloController.java
package com.example.springbootsnapshottest.controller;

import com.example.springbootsnapshottest.configuration.MultipleDataSourcesProperties;
import com.example.springbootsnapshottest.configuration.TestProperties;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("hello")
public class HelloController {
  private MultipleDataSourcesProperties multipleDataSourcesProperties;
  private TestProperties testProperties;

  public HelloController(MultipleDataSourcesProperties multipleDataSourcesProperties, TestProperties testProperties) {
    this.multipleDataSourcesProperties = multipleDataSourcesProperties;
    this.testProperties = testProperties;
  }

  @GetMapping
  public String index() {
    System.out.println(multipleDataSourcesProperties);
    System.out.println(testProperties);
    return "Hello";
  }
}

File: spring-kotlin-microservice/src/main/kotlin/com/example/demo/application/controller/HelloController.kt
package com.example.demo.application.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping()
class HelloController() {
    @GetMapping
    fun index(): String {
        return "Hello!"
    }

    @PostMapping
    fun create(): String {
        return "Hello!"
    }
}
